projects/table-builder/src/lib/directives/multi-sort.directive.ts

```diff
@@ -13,56 +14,33 @@ import { scan, filter, map } from 'rxjs/operators';
 })
 export class MultiSortDirective extends MatSort implements OnInit, OnDestroy {
   @Output() readonly multiSortChange: EventEmitter<Sort[]> = new EventEmitter<Sort[]>();
-
-  @Input() rules$: Observable<Sort[]> = new Subject();
-
-  @Input() set  reSort (val){
-    this.initRules();
-  } 
   rules: Sort[] = [];
-  preSortRules: Sort[] = [];
   private SubRef: Subscription;
 
+  constructor(private state: TableStateManager) {
+    super();
+  }
+
   ngOnInit() {
-    this.SubRef = this.rules$.pipe(
-      scan((acc,curr)=>({prev: acc.curr, curr }),{prev: null, curr: null} as {prev: Sort[], curr: Sort[]}),
-      filter(acc => !acc.prev || acc.curr.some(sort => !acc.prev.some(prevSort => sort.active === prevSort.active && sort.direction === prevSort.direction))),
-      map(acc => acc.curr)
+    this.SubRef = this.state.state$.pipe(
+      map( state => state.sorted ),
+      distinctUntilChanged(),
     ).subscribe( rules => {
-      this.preSortRules = rules;
-      this.initRules();
```

Comment:
The point of init rules was that when the table initializes an arrow should appear on that column.

Reply:
The arrows aren't really that helpful when multiple sorts are being done. It only really matters for when actually clicking the header to know what you did. So it's not necessary for initialization.